client/dashboard/domains/domain-diagnostics/index.tsx

```diff
@@ -0,0 +1,198 @@
+import Notice from '../../components/notice';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { Text } from '../../components/text';
+
+export default function DomainDiagnostics() {
+	const { domainName } = domainRoute.useParams();
+	const navigate = useNavigate();
+
+	const { data: domainDiagnostics, refetch: refetchDomainDiagnostics } = useSuspenseQuery(
+		domainDiagnosticsQuery( domainName )
+	);
+
+	const { mutate: fixDnsIssues, isPending: isFixing } = useMutation( {
+		...domainDnsEmailMutation( domainName ),
+		meta: {
+			snackbar: {
+				success: __( 'The default email DNS records were successfully restored.' ),
+				error: { source: 'server' },
+			},
+		},
+	} );
+
+	const { mutate: dismissNotice, isPending: isDismissing } = useMutation( {
+		...domainNoticeMutation( domainName, 'email-dns-records-diagnostics' ),
+		meta: {
+			snackbar: {
+				success: __( 'The domain diagnostics notice has been dismissed.' ),
+				error: { source: 'server' },
+			},
+		},
+	} );
+
+	const emailDnsDiagnostics = domainDiagnostics?.email_dns_records;
+
+	const renderDiagnosticForRecord = ( recordType: string ) => {
+		const uppercaseRecord = recordType.toUpperCase();
+		const record = emailDnsDiagnostics.records[ recordType ];
+
+		if ( ! record ) {
+			return null;
+		}
+
+		let message: string | null = null;
+
+		if ( record.error_message ) {
+			message = record.error_message;
+		} else if ( record.status === 'incorrect' ) {
+			message = sprintf(
+				/* translators: dnsRecordType is a DNS record type, e.g. SPF, DKIM or DMARC */
+				__( 'Your %(dnsRecordType)s record is incorrect. The correct record should be:' ),
+				{
+					dnsRecordType: uppercaseRecord,
+				}
+			);
+		} else if ( record.status === 'not_found' ) {
+			message = sprintf(
+				/* translators: dnsRecordType is a DNS record type, e.g. SPF, DKIM or DMARC */
+				__( 'There is no %(dnsRecordType)s record. The correct record should be:' ),
+				{
+					dnsRecordType: uppercaseRecord,
+				}
+			);
+		}
+
+		if ( ! message ) {
+			return null;
+		}
+
+		return (
+			<li key={ recordType } className="domain-diagnostics-notice__record-item">
+				<Text>{ message }</Text>
+				{ record.correct_record && (
+					<pre>
+						<code>{ record.correct_record }</code>
+					</pre>
+				) }
+			</li>
+		);
+	};
+
+	const renderDiagnostics = () => {
+		const emailDnsDiagnostics = domainDiagnostics?.email_dns_records;
+
+		if (
+			! emailDnsDiagnostics ||
+			emailDnsDiagnostics.code === 'domain_not_mapped_to_atomic_site' ||
+			emailDnsDiagnostics.all_essential_email_dns_records_are_correct
+		) {
+			return (
+				<Notice
+					variant="success"
+					title={ __( 'We didn’t find any issues with your domain!' ) }
+				></Notice>
+			);
+		}
+
+		const recordsToCheck = [ 'spf', 'dkim1', 'dkim2', 'dmarc' ];
+
+		return (
+			<VStack>
+				<Notice variant="error" title={ __( 'Missing or invalid DNS records' ) }>
```

Comment:
Let's add this notice as `<PageLayout>`'s `notices` prop so that it's consistent with the rest of MSD.